ozaria/site/components/cinematic/PageCinematic/index.vue

```diff
@@ -33,6 +48,13 @@ module.exports = Vue.extend({
   methods: {
     completedHandler () {
       this.$emit('completed')
+    },
+    async getCinematicData() {
+      try {
+        this.cinematicData = await getCinematic(this.cinematicIdOrSlug)
+      } catch (e) {
+        return Promise.reject(e)
```

Comment:
Each async function is [implicitly wrapped in a promise resolve](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#Rewriting_a_promise_chain_with_an_async_function) (see yellow block in that section).

`Promise.resolve(Promise.reject(e))` results in a rejected promise, which when received from an `await` call will throw.  Throwing `e` in this case will do the same thing, so we can actually remove the try / catch here and rely on async / await to handle this for us!

Reply:
Thanks for the link, it was really helpful.
Would it be the same thing, if I handle the error here instead of returning promise.reject (keeping the try/catch), and remove try catch from where I am calling this code?

Reply:
Yup!